src/app/components/planner/LabelLegend.tsx

```diff
@@ -0,0 +1,96 @@
+import { PlannerContext } from "@contexts/PlannerProvider";
+import { Label } from "@customTypes/Label";
+import { AccordionDetails, Button, Chip } from "@mui/joy";
+import Accordion, { accordionClasses } from "@mui/joy/Accordion";
+import AccordionSummary, {
+  accordionSummaryClasses,
+} from "@mui/joy/AccordionSummary";
+import { useContext, useState } from "react";
+
+import LabelsSelectionModal from "../modals/courseInfoModal/LabelSelectionModal";
+import CourseLabel from "../planner/quarters/courses/CourseLabel";
+
+export default function LabelLegend() {
+  const { getAllLabels, updatePlannerLabels } = useContext(PlannerContext);
+
+  const [showLabelSelectionModal, setShowLabelSelectionModal] = useState(false);
+
+  // Handlers
+  const handleEditLabels = () => {
+    setShowLabelSelectionModal(true);
+  };
+
+  const handleUpdateLabels = (labels: Label[]) => {
+    updatePlannerLabels({
+      labels,
+    });
+  };
+
+  return (
+    <div>
+      {getAllLabels() && (
+        <LabelsSelectionModal
```

Comment:
Don't render `LabelsSelectionModal`, but rather render the same content in this legend when edit mode is selected. This could mean abstracting some of the component logic to ensure DRY